make/go/dep_buf.mk

```diff
@@ -7,8 +7,8 @@ $(call _assert_var,CACHE_VERSIONS)
 $(call _assert_var,CACHE_BIN)
 
 # Settable
-# https://github.com/bufbuild/buf/releases 20240521 checked 20240524
-BUF_VERSION ?= v1.32.1
+# https://github.com/bufbuild/buf/releases 20240911 checked 20240916
+BUF_VERSION ?= v1.41.0
```

Comment:
FWIW we're on v1.42.0 now, but not that important.

Reply:
Yeah, this won't affect `bufbuild/buf`, but it's a good note to update `makego` for our other projects.